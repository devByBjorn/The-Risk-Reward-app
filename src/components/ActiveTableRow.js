import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
// import { calculatePositiveR, calculateNegativeR } from '../calculations/riskRewardCalculation'
import { EditIcon, TrashIcon } from '../icons/IconsComponents'
import { deleteFirebaseTrade } from '../actions/tradeActions'

const ActiveTableRow = (
  {
    dispatch,
    market,
    direction,
    setup,
    entry,
    stop,
    target,
    status,
    opened,
    negativeR,
    positiveR,
    id }) => (
    status === 'active' &&
    <tr>
      <td>{market}</td>
      <td>{direction}</td>
      <td>{setup}</td>
      <td>{entry}</td>
      <td>{stop}</td>
      <td>{target}</td>
      <td>{opened}</td>
      <td>{negativeR}</td> {/* loss -R*/}
      <td>{positiveR}</td> {/* win +R*/}
      <td>
        <div>
          <Link to={`/edit-trade/${id}`}>
            <EditIcon />
          </Link>
          <button onClick={() => {
            dispatch(deleteFirebaseTrade({ id }))
          }}
          ><TrashIcon />
          </button>
        </div>
      </td>
    </tr>
  )

export default connect()(ActiveTableRow)