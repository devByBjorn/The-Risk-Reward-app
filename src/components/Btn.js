import React from 'react'

export const Btn = (props) => (
  <button
    type="button"
    {...props}
  >
    {props.text}</button>
)

export const SubmitBtn = (props) => (
  <button>{props.text}</button>
)

export default Btn