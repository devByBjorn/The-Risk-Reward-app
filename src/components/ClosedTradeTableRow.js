import React, { useState, useEffect } from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { EditIcon, TrashIcon } from '../icons/IconsComponents'
import { deleteTrade } from '../actions/tradeActions'




const ClosedTradeTableRow = (
  {
    dispatch,
    market,
    direction,
    setup,
    opened,
    closed,
    status,
    outcome,
    rewardToRisk,
    id }) => {

  const [rowColor, setRowColor] = useState('')

  const colors = {
    win: '#7FBA00',
    loss: '#F0341B',
    scratch: '#FFB900'
  }
  const { win, loss, scratch } = colors

  useEffect(() => {
    if (outcome === 'win') {
      setRowColor(win)
    } else if (outcome === 'loss') {
      setRowColor(loss)
    } else {
      setRowColor(scratch)
    }
  })

  return (
    status === 'closed' &&

    <tr style={{ backgroundColor: rowColor }}>
      <td>{market}</td>
      <td>{direction}</td>
      <td>{setup}</td>
      <td>{rewardToRisk}</td>
      <td className="desktop">{opened}</td>
      <td className="desktop">{closed}</td>
      <td className="desktop">{outcome}</td>
      <td className="handle-trade-row">
        <div className="handle-trade">
          <Link to={`/edit-trade/${id}`}>
            <EditIcon
              className="edit-icon icon"
            />
          </Link>
          <button
            className="btn-transparent"
            onClick={() => {
              alert('Do you want to delete this trade?')
              dispatch(deleteTrade({ id }))
            }}
          ><TrashIcon
              className="trashcan-icon icon"
            />
          </button>
        </div>
      </td>
    </tr>
  )
}
export default connect()(ClosedTradeTableRow)