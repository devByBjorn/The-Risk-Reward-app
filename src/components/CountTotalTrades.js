import React from 'react'
import { connect } from 'react-redux'

const ClosedTradesTotal = ({ trades }) => (
  <h2>Total Closed Trades: {trades.length}</h2>
)

const mapStateToProps = (state) => {
  return {
    trades: state.trades.filter((trade) => trade.status === 'closed')
  }
}

export default connect(mapStateToProps)(ClosedTradesTotal)