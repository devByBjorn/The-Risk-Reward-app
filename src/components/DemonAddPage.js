import React from 'react'
import { connect } from 'react-redux'
import { NavLink } from 'react-router-dom'
import { addDemon } from '../actions/demonActions'
import DemonFinderForm from './demon-finder/DemonFinderForm'

const AddDemonPage = (props) => (
  <div>
    <h1>Add Demon Page</h1>
    <NavLink to="/demons">Back to demon list</NavLink>
    <div>
      <ol>
        <li>Give your demon suiting a title</li>
        <li>Think about your demon in terms of what,when,why,  how.
          <ul>
            <li>What makes the demon take over?</li>
            <li>When does this demon take over?</li>
            <li>Why...</li>
            <li>How can you manage this demon?</li>
          </ul>
        </li>
        <li>Write down your thoughts and process of managing this demon</li>
      </ol>
      <div>
        <h2>Demon</h2>
        <DemonFinderForm
          handleSubmit={(demon) => {
            props.dispatch(addDemon(demon))
            props.history.push('/demons')
          }}
        />
      </div>
    </div>
  </div>
)

export default connect()(AddDemonPage)