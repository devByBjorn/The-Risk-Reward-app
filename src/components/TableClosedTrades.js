import React, { useEffect, useState, useRef, Fragment, useMemo, useCallback } from 'react'
import { connect } from 'react-redux'
import { closedTradeSearch } from '../market-searcher/marketSearcher'
import TableEnhanced from './TableEnhanced'

const TableClosedTrades = ({ trades }) => {
  const [data, setData] = useState([])
  const [skipPageReset, setSkipPageReset] = useState(false)
  const fetchIdRef = useRef(0)

  const tableName = 'Closed Trades'

  const formDate = (milliSeconds) => {
    const date = new Date(milliSeconds)
    const months = ['Jan', 'Feb', 'Mar', 'Apr', 'May',
      'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
    const fullYear = date.getFullYear()
    const month = months[date.getMonth()]
    const dayDate = date.getDate()

    return `${dayDate} ${month}, ${fullYear}`
  }
  const columns = useMemo(
    () => [
      {
        Header: 'Market',
        accessor: 'market',
      }, {
        Header: 'Direction',
        accessor: 'direction'
      },
      {
        Header: 'Setup',
        accessor: 'setup',
      },
      {
        Header: 'R',
        accessor: 'rMultiple'
      },
      {
        Header: 'Opened',
        accessor: 'opened',
        Cell: (props) => {
          const customDate = formDate(props.value)
          return (
            <span>{customDate}</span>
          )
        }
      },
      {
        Header: 'Closed',
        accessor: 'closed',
        Cell: (props) => {
          const customDate = formDate(props.value)
          return (
            <span>{customDate}</span>
          )
        }
      },
      {
        Header: 'Outcome',
        accessor: 'outcome',
      },
    ],
    []
  )

  useEffect(() => {
    setData(trades)
  }, [trades])

  const fetchData = useCallback(() => {
    const fetchId = ++fetchIdRef.current
    if (fetchId === fetchIdRef.current) {
      setData(trades)
    }
  }, [])

  const updateMyData = (rowIndex, columnId, value) => {
    // We also turn on the flag to not reset the page
    setSkipPageReset(true)
    setData(old =>
      old.map((row, index) => {
        if (index === rowIndex) {
          return {
            ...old[rowIndex],
            [columnId]: value,
          }
        }
        return row
      })
    )
  }

  return (
    <div>
      <TableEnhanced
        tableName={tableName}
        columns={columns}
        data={data}
        fetchData={fetchData}
        setData={setData}
        updateMyData={updateMyData}
        skipPageReset={skipPageReset}
      />
    </div>
  )
}

const mapStateToProps = (state) => ({
  trades: closedTradeSearch(state.trades, state.filters)
})

export default connect(mapStateToProps)(TableClosedTrades)