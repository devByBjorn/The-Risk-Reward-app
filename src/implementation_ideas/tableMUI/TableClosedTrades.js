import React, { useEffect, useState, useRef, Fragment, useMemo, useCallback } from 'react'
import { connect } from 'react-redux'
import { closedTradeSearch } from '../../market-searcher/marketSearcher'
import CssBaseline from '@material-ui/core/CssBaseline'
import EnhancedTable from './EnhancedTable'


const TableClosedTrades = ({ trades }) => {

  const [data, setData] = useState([])
  const [skipPageReset, setSkipPageReset] = useState(false)
  const fetchIdRef = useRef(0)

  console.log('ALL', trades)
  console.log('CLOSED', trades.filter((trade) => trade.status === 'closed'))

  const columns = useMemo(
    () => [
      {
        Header: 'Market',
        accessor: 'market',
        className: 'market-td'
      }, {
        Header: 'Direction',
        accessor: 'direction'
      },
      {
        Header: 'Setup',
        accessor: 'setup',
      },
      {
        Header: 'Status',
        accessor: 'status',
      },
      {
        Header: 'R',
        accessor: 'rewardToRisk'
      },
      {
        Header: 'Opened',
        accessor: 'opened',
      },
      {
        Header: 'Closed',
        accessor: 'closed'
      },
      {
        Header: 'Outcome',
        accessor: 'outcome',
      },
    ],
    []
  )

  useEffect(() => {
    setData(trades)
  }, [trades])

  const fetchData = useCallback(() => {
    const fetchId = ++fetchIdRef.current
    if (fetchId === fetchIdRef.current) {
      setData(trades)
    }
  }, [])

  const updateMyData = (rowIndex, columnId, value) => {
    // We also turn on the flag to not reset the page
    setSkipPageReset(true)
    setData(old =>
      old.map((row, index) => {
        if (index === rowIndex) {
          return {
            ...old[rowIndex],
            [columnId]: value,
          }
        }
        return row
      })
    )
  }

  return (
    <div>
      <CssBaseline />
      <h2>Closed Trades</h2>
      <EnhancedTable
        columns={columns}
        data={data}
        fetchData={fetchData}
        setData={setData}
        updateMyData={updateMyData}
        skipPageReset={skipPageReset}
      />
    </div>
  )
}

const mapStateToProps = (state) => ({
  trades: closedTradeSearch(state.trades, state.filters)
})

export default connect(mapStateToProps)(TableClosedTrades)