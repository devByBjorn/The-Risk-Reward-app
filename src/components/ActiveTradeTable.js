import React from 'react'
import { connect } from 'react-redux'
import ActiveTableRow from './ActiveTableRow'
import { ThMobileAndDesktop, ThOnlyDesktop } from './TradeTableHead'
const ActiveTradeTable = ({ trades }) => (
  <div>
    <table className="table-grid active">
      <caption>Active Trades</caption>
      <thead>
        <tr>
          <ThMobileAndDesktop heading='Market' />
          <ThMobileAndDesktop heading='Direction' />
          <ThMobileAndDesktop heading='Setup' />
          <ThMobileAndDesktop heading='Entry' />
          <ThMobileAndDesktop heading='Stop' />
          <ThMobileAndDesktop heading='Target' />
          <ThOnlyDesktop heading='Opened' />
          <ThMobileAndDesktop heading='-R' />
          <ThMobileAndDesktop heading='+R' />
        </tr>
      </thead>
      <tbody>
        {trades.map((trade) =>
          (<ActiveTableRow
            {...trade}
            key={trade.id}
          />)
        )}
      </tbody>
    </table>
  </div>
)

const mapStateToProps = (state) => {
  return {
    trades: state.trades,
  }
}

export default connect(mapStateToProps)(ActiveTradeTable)