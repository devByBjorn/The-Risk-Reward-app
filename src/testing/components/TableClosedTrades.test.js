import React from 'react'
import { shallow } from 'enzyme'
import { trades } from '../fixtures/trades'
import { TableClosedTrades } from '../../components/TableClosedTrades'

test('should render TabelClosedTrades without data', () => {
  const wrapper = shallow(<TableClosedTrades trades={[]} />)
  expect(wrapper).toMatchSnapshot()
})
test('should render TabelClosedTrades with trades data', () => {
  const wrapper = shallow(<TableClosedTrades trade={trades[0]} />)
  expect(wrapper).toMatchSnapshot()
})