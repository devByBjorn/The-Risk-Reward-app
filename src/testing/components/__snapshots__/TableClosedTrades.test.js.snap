// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TabelClosedTrades with trades data 1`] = `
<div>
  <Connect(EnhancedTable)
    columns={
      Array [
        Object {
          "Header": "Market",
          "accessor": "market",
        },
        Object {
          "Header": "Direction",
          "accessor": "direction",
        },
        Object {
          "Header": "Setup",
          "accessor": "setup",
        },
        Object {
          "Header": "R",
          "accessor": "rMultiple",
        },
        Object {
          "Cell": [Function],
          "Header": "Opened",
          "accessor": "opened",
        },
        Object {
          "Cell": [Function],
          "Header": "Closed",
          "accessor": "closed",
        },
        Object {
          "Header": "Outcome",
          "accessor": "outcome",
        },
      ]
    }
    data={Array []}
    fetchData={[Function]}
    setData={[Function]}
    skipPageReset={false}
    tableName="Closed Trades"
    updateMyData={[Function]}
  />
</div>
`;

exports[`should render TabelClosedTrades without data 1`] = `
<div>
  <Connect(EnhancedTable)
    columns={
      Array [
        Object {
          "Header": "Market",
          "accessor": "market",
        },
        Object {
          "Header": "Direction",
          "accessor": "direction",
        },
        Object {
          "Header": "Setup",
          "accessor": "setup",
        },
        Object {
          "Header": "R",
          "accessor": "rMultiple",
        },
        Object {
          "Cell": [Function],
          "Header": "Opened",
          "accessor": "opened",
        },
        Object {
          "Cell": [Function],
          "Header": "Closed",
          "accessor": "closed",
        },
        Object {
          "Header": "Outcome",
          "accessor": "outcome",
        },
      ]
    }
    data={Array []}
    fetchData={[Function]}
    setData={[Function]}
    skipPageReset={false}
    tableName="Closed Trades"
    updateMyData={[Function]}
  />
</div>
`;
